# Expenses in Highlands Sheet are incorrect
* Using expenses from Data link to fill in expenditures for March 25 presentation


In [1]:
import numpy as np
import pandas as pd
import TIRCP_functions
from siuba import *
from calitp import *

GCS_FILE_PATH = "gs://calitp-analytics-data/data-analyses/tircp/"

pd.options.display.max_columns = 50
pd.options.display.max_rows = 120
pd.set_option('display.max_colwidth', None)

/opt/conda/lib/python3.9/site-packages/geopandas/_compat.py:111: UserWarning: The Shapely GEOS version (3.10.2-CAPI-1.16.0) is incompatible with the GEOS version PyGEOS was compiled with (3.10.1-CAPI-1.16.0). Conversions between both will be slow.
E0314 21:34:51.905357170    1933 fork_posix.cc:70]           Fork support is only compatible with the epoll1 and poll polling strategies
E0314 21:34:52.460633708    1933 fork_posix.cc:70]           Fork support is only compatible with the epoll1 and poll polling strategies


## Expenditure information from the Enterprise Data Link data Linda pulled.

In [2]:
enterprise = "Enterprise.xlsx"
enterprise_df = pd.read_excel(f"{GCS_FILE_PATH}{enterprise}")
enterprise_df = to_snakecase(enterprise_df)

In [69]:
enterprise_df.head(2)

fy cycle  project_id                                project_name  \
0  2016   One    16000007  Refurbishment of Seven Light Rail Vehicles   
1  2016   One    16000008   South Bay Bus Rapid Transit (BRT) Project   

     tot_exp  
0 2182302.44  
1   30770.46

In [4]:
enterprise_df = enterprise_df.rename(columns = {'project':'project_id'})

In [5]:
#keep only columns necessary.
enterprise_df = enterprise_df[['fy','cycle','project_id','project_name','tot_exp']]

In [6]:
#Looking at IDS by cycles.
enterprise_df.cycle.value_counts()

Three    152
One       47
Two       47
Four       8
Name: cycle, dtype: int64

In [7]:
enterprise_df.shape

(254, 5)

#### Aggregate by project ID and each cycle, so each project has only one record for total expenses

In [8]:
enterprise_df_project_id = enterprise_df.groupby(['project_id','cycle']).agg({'tot_exp':'sum'}).reset_index()

In [9]:
enterprise_df_project_id.shape

(106, 3)

In [10]:
enterprise_df_project_id.head(2)

project_id cycle     tot_exp
0    16000007   One  6315208.89
1    16000008   One  4000000.00

## Merge in with allocation sheet of Excel workbook 
* There are 171 unique project IDS in our allocation sheet compared to 106 project IDS in Data Link....
* Lots of missing Project IDS: 107 out of 307 rows of data.

In [11]:
#read in allocation df because that's the sheet with project ids. 
allocation = TIRCP_functions.allocation()

In [12]:
allocation.shape

(215, 50)

In [13]:
allocation.isna().sum()

Award_Year                                  2
Project_#                                   2
Grant_Recipient                             2
Implementing_Agency                         2
Project_ID                                  7
EA                                         10
Components                                  2
Phase                                       2
Allocation_Amt_Allocation_Sheet             0
Expended_Amt_Allocation_Sheet               0
SB1_Funding                                 1
SB1_Budget_Year                            60
GGRF_Funding                                0
GGRF_Budget_Year                           43
CTC_Financial_Resolution                    9
CTC_Allocation_Amendment                  186
CTC_Waiver                                208
CalSTA_Waiver                             197
PSA_#                                       8
CT_Document_#                              33
Date_Branch_Chief_Receives_PSA            209
Date_Regional_Coordinator_Receives

In [14]:
#keep only columns that are relevant.
allocation_df = allocation[['Award_Year','Project_ID','PPNO']]

In [15]:
f'There are {allocation_df.Project_ID.nunique()} unique project ids in allocation'

'There are 191 unique project ids in allocation'

In [16]:
#only keep rows that have Project IDS filled in
allocation_df = allocation_df.loc[allocation_df['Project_ID'].notnull()]

In [17]:
allocation_df.shape

(208, 3)

In [18]:
joined_alloc = pd.merge(allocation_df, enterprise_df_project_id, 
                        left_on='Project_ID', right_on='project_id', 
                        how = 'outer', indicator = True)

In [19]:
joined_alloc._merge.value_counts()

both          119
left_only      89
right_only      1
Name: _merge, dtype: int64

In [74]:
joined_alloc.sample(10)

Award_Year     Project_ID   PPNO  project_id  cycle     tot_exp  \
44      2016.00       18000237  CP017 18000237.00    Two 12705054.75   
90      2018.00       19000071  CP042         NaN    NaN         NaN   
58      2018.00       20000185  CP028         NaN    NaN         NaN   
25      2016.00       21000266  CP076         NaN    NaN         NaN   
164     2018.00   \n0019000457  CP033         NaN    NaN         NaN   
177     2018.00       21000200  CP033         NaN    NaN         NaN   
19      2016.00       17000181  CP023         NaN    NaN         NaN   
59      2018.00       19000064  CP028 19000064.00  Three    71539.79   
165     2018.00       20000207  CP033         NaN    NaN         NaN   
97      2018.00       19000294  CP042         NaN    NaN         NaN   

        _merge  
44        both  
90   left_only  
58   left_only  
25   left_only  
164  left_only  
177  left_only  
19   left_only  
59        both  
165  left_only  
97   left_only

In [21]:
#only keep "both" results...
joined_alloc2 = joined_alloc.loc[joined_alloc['_merge'] == 'both'].drop(columns =['_merge'])

In [22]:
#group by so that only one row for each PPNO number & Year
joined_alloc3 = (joined_alloc2
                  .groupby(['PPNO', 'Award_Year'])
                  .agg({'tot_exp':'sum'})
                  .reset_index()
                 )

In [23]:
joined_alloc3.shape

(51, 3)

In [24]:
#rename cols before joining with "joined" df 
joined_alloc3 = joined_alloc3.add_prefix('from_joined_alloc_df_')

In [25]:
joined_alloc3

from_joined_alloc_df_PPNO  from_joined_alloc_df_Award_Year  \
0                       1155                           2018.0   
1                       1230                           2016.0   
2                      2320B                           2018.0   
3                      CP001                           2015.0   
4                      CP002                           2015.0   
5                      CP003                           2015.0   
6                      CP004                           2015.0   
7                      CP005                           2015.0   
8                      CP006                           2015.0   
9                      CP006                           2016.0   
10                     CP006                           2018.0   
11                     CP007                           2015.0   
12                     CP008                           2015.0   
13                     CP010                           2015.0   
14                     CP011                           2015.0   
15                     CP012                           2015.0   
16                     CP013                           2015.0   
17                     CP014                           2015.0   
18                     CP015                           2015.0   
19                     CP016                           2016.0   
20                     CP017                           2016.0   
21                     CP018                           2016.0   
22                     CP019                           2016.0   
23                     CP020                           2016.0   
24                     CP021                           2016.0   
25                     CP022                           2016.0   
26                     CP023                           2016.0   
27                     CP024                           2016.0   
28                     CP025                           2016.0   
29                     CP027                           2018.0   
30                     CP028                           2018.0   
31                     CP031                           2018.0   
32                     CP032                           2018.0   
33                     CP033                           2018.0   
34                     CP034                           2018.0   
35                     CP035                           2018.0   
36                     CP036                           2018.0   
37                     CP039                           2018.0   
38                     CP041                           2018.0   
39                     CP043                           2016.0   
40                     CP046                           2018.0   
41                     CP047                           2018.0   
42                     CP048                           2018.0   
43                     CP051                           2018.0   
44                     CP053                           2018.0   
45                     CP055                           2018.0   
46                     CP061                           2020.0   
47                     CP062                           2020.0   
48                     CP069                           2020.0   
49                     CP072                           2020.0   
50                     CP079                           2018.0   

    from_joined_alloc_df_tot_exp  
0                      486509.70  
1                     9204000.00  
2                      500000.00  
3                     6315208.89  
4                    38023039.68  
5                     4000000.00  
6                     2320000.00  
7                    22284205.53  
8                    41181000.00  
9                    45092000.00  
10                   26867000.00  
11                     277840.47  
12                   31936000.00  
13                   11000000.00  
14                    6841000.00  
15                    4619999.90  
16                   100000

In [26]:
joined_alloc3.astype({'from_joined_alloc_df_Award_Year': 'int64'}).dtypes

from_joined_alloc_df_PPNO           object
from_joined_alloc_df_Award_Year      int64
from_joined_alloc_df_tot_exp       float64
dtype: object

## Joining original sheets on Award Year & PPNO

In [75]:
project = TIRCP_functions.project()

In [78]:
project.isna().sum()

Award_Year                                         0
Project_#                                          0
Grant_Recipient                                    0
Project_Title                                      0
District                                           3
County                                             1
Project_Description                                0
Master_Agreement_Number                            2
Master_Agreement_Expiration_Date                   2
Project_Manager                                    0
Regional_Coordinator                               0
Technical_Assistance-CALITP_(Y/N)                 10
Technical_Assistance-Fleet_(Y/N)                  12
Technical_Assistance-Network_Integration_(Y/N)    11
Technical_Assistance-Priority_Population_(Y/N)    11
Total_Project_Cost                                 0
TIRCP_project_sheet                                0
Allocated_Amount                                   0
Unallocated_amt_project_sheet                 

In [31]:
final_join = pd.merge(project, joined_alloc3, 
                      left_on = ['PPNO', 'Award_Year'], 
                      right_on = ['from_joined_alloc_df_PPNO', 'from_joined_alloc_df_Award_Year'],
                      how = 'left', indicator = True)

In [32]:
final_join._merge.value_counts()

both          50
left_only     25
right_only     0
Name: _merge, dtype: int64

#### Check to see if PPNO matches  cross original project sheet & joined 

In [33]:
final_join.shape

(75, 36)

In [34]:
final_join.PPNO.nunique()

69

In [35]:
project.PPNO.nunique()

69

In [36]:
PPNO_project = set(project.PPNO.unique().tolist())
PPNO_join = set(final_join.PPNO.unique().tolist())
PPNO_join - PPNO_project 

set()

### Checking out our 2 columns of expended amounts to make sure Data Link information makes sense.
* Expended_Amt_Project_sheet is from Highlands sheet.
* from_joined_alloc_df_tot_exp from Data Link

In [37]:
pd.options.display.float_format = '{:,}'.format
pd.options.display.float_format = "{:.2f}".format

In [38]:
final_join[['Award_Year','PPNO','Project_Title', 'TIRCP_project_sheet', 'Expended_Amt_project_sheet', 'from_joined_alloc_df_tot_exp', '_merge']].sort_values('PPNO')

Award_Year   PPNO  \
57        2018   1155   
24        2016   1230   
29        2018  2320B   
6         2015  CP001   
12        2015  CP002   
7         2015  CP003   
5         2015  CP004   
0         2015  CP005   
25        2016  CP006   
47        2018  CP006   
9         2015  CP006   
3         2015  CP007   
8         2015  CP008   
13        2015  CP010   
11        2015  CP011   
1         2015  CP012   
4         2015  CP013   
10        2015  CP014   
2         2015  CP015   
17        2016  CP016   
21        2016  CP017   
14        2016  CP019   
19        2016  CP020   
22        2016  CP022   
15        2016  CP023   
26        2016  CP025   
28        2016  CP026   
30        2018  CP027   
31        2018  CP028   
32        2018  CP028   
36        2018  CP029   
39        2018  CP030   
40        2018  CP031   
41        2018  CP031   
56        2018  CP033   
46        2018  CP034   
44        2018  CP034   
48        2018  CP035   
34        2018  CP036   
42        2018  CP039   
55        2018  CP041   
20        2016  CP043   
53        2018  CP045   
54        2018  CP046   
51        2018  CP047   
50        2018  CP048   
38        2018  CP051   
43        2018  CP053   
37        2018  CP054   
33        2018  CP055   
52        2018  CP057   
27        2016  CP057   
18        2016  CP058   
58        2020  CP059   
59        2020  CP060   
60        2020  CP061   
61        2020  CP062   
62        2020  CP063   
63        2020  CP064   
64        2020  CP065   
65        2020  CP066   
66        2020  CP067   
67        2020  CP068   
68        2020  CP069   
69        2020  CP070   
70        2020  CP071   
71        2020  CP072   
72        2020  CP073   
73        2020  CP074   
74        2020  CP075   
16        2016  CP076   
45        2018  CP077   
49        2018  CP078   
35        2018  CP079   
23        2016  CP080   

                                                                                                           Project_Title  \
57                                                                                Extend rail service to Monterey County   
24                                                                                       Redlands Passenger Rail Project   
29                      Purchase Zero Emission High Capacity Buses to Support Transbay Tomorrow and Clean Corridors Plan   
6                                                   Sacramento Regional Transit's Refurbishment of 7 Light Rail Vehicles   
12                                                         Purchase of Nine Fuel-Efficient Tier IV EMD F-125 Locomotives   
7                                                                                            South Bay Bus Rapid Transit   
5                                                                                           Bravo! Route 560 Rapid Buses   
0                                                         Regional Transit Interconnectivity & Environmental Sustability   
25                                                                        Light Rail Modernization and Expansion Program   
47                                                                                    Transit Capacity Expansion Program   
9                                                                               SFMTA Light Rail Vehicle Fleet Expansion   
3                                                                             Pacific Surfliner Transit Transfer Program   
8                                                    San Diego Metropolitan Transit System Trolley Capacity Improvements   
13                                                                                       SMART Rail Car Capacity Project   
11                                          Bus Rapid Transit – Martin Luther King Corridor and Crosstown Miner Corridor   
1                                                                                          Travel Time Reduction Projec

In [39]:
#fill in with 0
final_join['from_joined_alloc_df_tot_exp'] = final_join['from_joined_alloc_df_tot_exp'].fillna(0)

In [102]:
def tableau2(df):
    #Keeping only the columns we want
    df = (df[['PPNO','Award_Year', 'Project_#', 'Grant_Recipient', 
       'Project_Title', 'District', 'County', 'Project_Description',
       'Master_Agreement_Number', 'Master_Agreement_Expiration_Date',
       'Project_Manager', 'Regional_Coordinator', 'Technical_Assistance-Fleet_(Y/N)',
       'Technical_Assistance-Network_Integration_(Y/N)',
       'Technical_Assistance-Priority_Population_(Y/N)', 'Total_Project_Cost',
       'TIRCP_project_sheet', 'Allocated_Amount',
       'Percentage_Allocated',
       'from_joined_alloc_df_tot_exp', 'Other_Funds_Involved']]
                 )
    #Getting percentages & filling in with 0
    df['Expended_Percent'] = df['from_joined_alloc_df_tot_exp']/df['Allocated_Amount']
    df['Allocated_Percent'] = df['Allocated_Amount']/df['TIRCP_project_sheet']
    
    # Subtract TIRCP with Allocated Amount with Unallocated
    df['Unallocated_Amount'] = df["TIRCP_project_sheet"] - df["Allocated_Amount"]
    #filling in for 0's 
    df[['Expended_Percent','Allocated_Percent']] = df[['Expended_Percent','Allocated_Percent']].fillna(value=0)
   
    df[['Expended_Percent','Allocated_Percent']]  =  df[['Expended_Percent','Allocated_Percent']].replace(np.inf, 0)
    #Categorizing expended percentage into bins
    def expended_percent(row):
           
            if ((row.Expended_Percent > 0) and (row.Expended_Percent < .26)):
                return "1-25"
            elif ((row.Expended_Percent > .25) and (row.Expended_Percent < .51)):
                return "26-50"
            elif ((row.Expended_Percent > .50) and (row.Expended_Percent < .76)):
                return "51-75"
            elif ((row.Expended_Percent > .75) and (row.Expended_Percent < 1.0)):
                return "76-99"
            elif row.Expended_Percent == 0.0:
                return "0"
            else:
                return "100"
    df["Expended_Percent_Group"] = df.apply(lambda x: expended_percent(x), axis=1)
    
    # Categorize years and expended_percent_group into bins
    def progress(df):   
        ### 2015 ### 
        if (df['Award_Year'] == 2015) and (df['Expended_Percent_Group'] == "1-25") | (df['Expended_Percent_Group'] == "26-50"):
            return 'Behind'
        elif (df['Award_Year'] == 2015) and (df['Expended_Percent_Group'] == "76-99") | (df['Expended_Percent_Group'] == "51-75"):
            return 'On Track'
        
        ### 2016 ###
        elif (df['Award_Year'] == 2016) and (df['Expended_Percent_Group'] == "1-25") | (df['Expended_Percent_Group'] == "26-50"):
            return 'Behind'
        elif (df['Award_Year'] == 2016) and (df['Expended_Percent_Group'] == "51-75") | (df['Expended_Percent_Group'] == "76-99"):
            return 'On Track'
        
        ### 2018 ###
        elif (df['Award_Year'] == 2018) and (df['Expended_Percent_Group'] == "1-25"):
            return 'Behind'
        elif (df['Award_Year'] == 2018) and (df['Expended_Percent_Group'] == "26-50") | (df['Expended_Percent_Group'] == "51-75"):
            return 'On Track'
        elif (df['Award_Year'] == 2018) and (df['Expended_Percent_Group'] == "76-99"):
            return 'Ahead'
        
        ### 2020 ### 
        elif (df['Award_Year'] == 2020) and (df['Expended_Percent_Group'] == "1-25"):
            return 'Behind'
        elif (df['Award_Year'] == 2020) and (df['Expended_Percent_Group'] == "26-50"):
            return 'On Track'
        elif (df['Award_Year'] == 2020) and (df['Expended_Percent_Group'] == "51-75") | (df['Expended_Percent_Group'] == "76-99"):
            return 'Ahead'
        
        ### 0 Expenditures ### 
        elif df ['Expended_Percent_Group'] == "0":
            return "No expenditures recorded"
        
        ### Else ### 
        else: 
            return "100% of allocated funds spent"

    df['Progress'] = df.apply(progress, axis = 1)
    
    #Which projects are large,small, medium
    p75 = df.TIRCP_project_sheet.quantile(0.75).astype(float)
    p25 = df.TIRCP_project_sheet.quantile(0.25).astype(float)
    p50 = df.TIRCP_project_sheet.quantile(0.50).astype(float)
    
    def project_size (row):
        if ((row.TIRCP_project_sheet > 0) and (row.TIRCP_project_sheet < p25)):
            return "Small"
        elif ((row.TIRCP_project_sheet > p25) and (row.TIRCP_project_sheet < p75)):
            return "Medium"
        elif ((row.TIRCP_project_sheet > p50) and (row.TIRCP_project_sheet > p75 )):
            return "Large"
        else:
            return "$0 recorded for TIRCP"
        
    df["Project_Category"] = df.apply(lambda x: project_size(x), axis=1)
     #Rename cols to the right names
    df = (df.rename(columns = {'from_joined_alloc_df_tot_exp':'Expended_Amount', 
                                                'TIRCP_project_sheet': "TIRCP_Amount"}
                  ))
    ### GCS ###
   # df = df.to_excel(f'{GCS_FILE_PATH}INTERIM_EXPENDITURE_Tableau_Parquet.xlsx')
  
    return df 

In [98]:
final_df = tableau2(final_join)

/tmp/ipykernel_1933/403944568.py:14: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
/tmp/ipykernel_1933/403944568.py:15: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
/tmp/ipykernel_1933/403944568.py:18: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
/tmp/ipykernel_1933/40394

In [116]:
final_df[['Award_Year','PPNO','District','Project_Title', 'TIRCP_Amount','Allocated_Amount','Expended_Amount','Expended_Percent','Progress']].sort_values('Expended_Percent')

Award_Year   PPNO District  \
37        2018  CP054        7   
28        2016  CP026       10   
36        2018  CP029        7   
73        2020  CP074       10   
39        2018  CP030        7   
45        2018  CP077       11   
49        2018  CP078        4   
52        2018  CP057      NaN   
53        2018  CP045        2   
58        2020  CP059        7   
59        2020  CP060      NaN   
62        2020  CP063        1   
63        2020  CP064        7   
64        2020  CP065        7   
65        2020  CP066      VAR   
66        2020  CP067        3   
67        2020  CP068        8   
69        2020  CP070        4   
70        2020  CP071      NaN   
72        2020  CP073        7   
27        2016  CP057        4   
23        2016  CP080        3   
74        2020  CP075        4   
16        2016  CP076        7   
18        2016  CP058        7   
60        2020  CP061        3   
35        2018  CP079        6   
20        2016  CP043      VAR   
38        2018  CP051        4   
26        2016  CP025      VAR   
29        2018  2320B        4   
33        2018  CP055        4   
56        2018  CP033        7   
15        2016  CP023      VAR   
68        2020  CP069       11   
3         2015  CP007      VAR   
48        2018  CP035      VAR   
71        2020  CP072        4   
14        2016  CP019        7   
41        2018  CP031      VAR   
32        2018  CP028        7   
31        2018  CP028        7   
50        2018  CP048        5   
42        2018  CP039        4   
43        2018  CP053        3   
17        2016  CP016        6   
30        2018  CP027       12   
19        2016  CP020        7   
34        2018  CP036        4   
21        2016  CP017       12   
46        2018  CP034       13   
40        2018  CP031      VAR   
61        2020  CP062        7   
51        2018  CP047        5   
54        2018  CP046        4   
44        2018  CP034        8   
22        2016  CP022        4   
0         2015  CP005        7   
55        2018  CP041        4   
12        2015  CP002        7   
57        2018   1155      VAR   
6         2015  CP001        3   
1         2015  CP012        4   
24        2016   1230        8   
2         2015  CP015        7   
25        2016  CP006        4   
4         2015  CP013        5   
47        2018  CP006        4   
7         2015  CP003       11   
8         2015  CP008       11   
10        2015  CP014       10   
11        2015  CP011       10   
13        2015  CP010        4   
5         2015  CP004       12   
9         2015  CP006        4   

                                                                                                           Project_Title  \
37                                                 Electric Blue: Electrification of City of Santa Monica's Big Blue Bus   
28                                                                          SB 132 ACE Extension Lathrop to Ceres/Merced   
36                            Los Angeles City: Leading the Transformation to Zero-Emission Electric Bus Transit Service   
73                 Improving Air Quality & Economic Growth with Electric Buses in Merced County, the Gateway to Yosemite   
39                                   Los Angeles Region Transit System Integration and Modernization Program of Projects   
45                                                       Ride Between the Line: Enhancing Access to Transit in San Diego   
49                                                                                            SamTrans Express Bus Pilot   
52                                                                         VTA’s BART Silicon Valley Extension, Phase II   
53                                                                                      North State Intercity Bus System   
58                    Reaching the Most Transit-Vulnerable: AVTA's Zero Emission "Microtransit" & Bus Expansion Proposal   
59                                                     

In [105]:
final_df.Progress.value_counts()

No expenditures recorded         25
On Track                         19
Behind                           14
100% of allocated funds spent    12
Ahead                             5
Name: Progress, dtype: int64

In [115]:
final_df.isna().sum()

PPNO                                               0
Award_Year                                         0
Project_#                                          0
Grant_Recipient                                    0
Project_Title                                      0
District                                           3
County                                             1
Project_Description                                0
Master_Agreement_Number                            2
Master_Agreement_Expiration_Date                   2
Project_Manager                                    0
Regional_Coordinator                               0
Technical_Assistance-Fleet_(Y/N)                  12
Technical_Assistance-Network_Integration_(Y/N)    11
Technical_Assistance-Priority_Population_(Y/N)    11
Total_Project_Cost                                 0
TIRCP_Amount                                       0
Allocated_Amount                                   0
Percentage_Allocated                          

In [106]:
TIRCP = final_df['TIRCP_Amount'].sum()
f'The total amount of TIRCP is {TIRCP}'

'The total amount of TIRCP is 5837722000'

In [107]:
Alloc = final_df['Allocated_Amount'].sum()
f'The total amount of $ allocated is {Alloc}'

'The total amount of $ allocated is 2283509000'

In [108]:
Exp = final_df['Expended_Amount'].sum()
f'Total amount of expenditures is {Exp}'

'Total amount of expenditures is 749827692.08'

In [109]:
f'% of allocated money spent {Alloc/TIRCP}'

'% of allocated money spent 0.3911643959750053'

In [110]:
f'% of expended  by allocation {Exp/Alloc}'

'% of expended  by allocation 0.3283664273186574'

In [111]:
f'% of expended  by TIRCP {Exp/TIRCP}'

'% of expended  by TIRCP 0.1284452552005731'

In [113]:
#final_df.to_excel(f'{GCS_FILE_PATH}tableau_with_temporary_expenditure_sol.xlsx', index = False)